Classes/Capture/AddNoteViewController.swift

```diff
@@ -54,12 +54,114 @@ final class AddNoteViewController: UIViewController {
+        let urlButton = UIBarButtonItem( customView: button1 )
+        let button2 = UIButton( type: .custom )
+        button2.setImage( UIImage( named: "datepick.png" ), for: .normal )
+        button2.addTarget(self, action: #selector(setDatePickerView), for: .touchUpInside)
+        button2.frame = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        button2.bounds = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        let dateButton = UIBarButtonItem( customView: button2 )
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        bar.items = [urlButton, dateButton, spacer ]
+        bar.sizeToFit()
+        return bar
+    }()
+
+    @objc func addLinkMarkup() {
+        guard self.textView.text != nil else { return } // If no selected text, ignore.  Consider popup?
+        if let range = self.textView.selectedTextRange, !range.isEmpty {
+            let selectedText = self.textView.text( in: range )
+            self.textView.replace( range, withText: "[[\(selectedText!)][]]" )
+            let newCursor = self.textView.position(from: range.end, offset: 4)!
+            self.textView.selectedTextRange = self.textView.textRange( from: newCursor, to: newCursor)
+        }
+    }
+
+    let datePicker: UIDatePicker = {
+        let picker = UIDatePicker(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 216))
+        picker.datePickerMode = .dateAndTime
+        return picker
+    }()
+
+    private lazy var datePickBar: UIToolbar = {
+        let bar = UIToolbar()
+        let cancelButton = UIBarButtonItem( title: "Cancel", style: .plain, target: self, action: #selector(setDefaultInputView) )
+        let scheduleButton = UIBarButtonItem( title: "Schd", style: .plain, target: self, action: #selector(insertDateSchedule) )
+        let deadlineButton = UIBarButtonItem( title: "Dead", style: .plain, target: self, action: #selector(insertDateDeadline) )
+        let agendaButton = UIBarButtonItem( title: "Agenda", style: .plain, target: self, action: #selector(insertDateAgenda) )
+        let plainButton = UIBarButtonItem( title: "Plain", style: .plain, target: self, action: #selector(insertDatePlain) )
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        bar.items = [ scheduleButton, deadlineButton, agendaButton, plainButton, spacer, cancelButton ]
+        bar.sizeToFit()
+        return bar
+    }()
+
+    private func datePicked() -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "YYYY-MM-dd E hh:mm"
+        //dateFormatter.formatOptions = [.withTime]
+        return dateFormatter.string( from: self.datePicker.date )
+    }
+
+    private func replaceSelected( with newString: String ) {
+        print( "replaceSelected" )
+        if self.textView.text != nil {
+            if let range = self.textView.selectedTextRange {
```

Comment:
In Swift, usually, such statements are written with the `guard` statement. Like this:
```swift
guard let range = self.textView.selectedTextRange, !range.isEmpty else { return }
self.textView.replace( range, withText: newString )
```

Reply:
I don't understand the difference here.  Mind explaining?  Is it just so that the code which is the primary function isn't indented in the body of the if?

Reply:
It's more of a style consistency and clarity thing than just functional. I think this explains it nicely: https://thatthinginswift.com/guard-statement-swift/  and I especially like that it calls out the danger of "combining tests for multiple values". 

I do prefer `guard` to `if let` nesting. In this case, you don't have a bunch of nesting, but for consistency in general `guard` can still be used.

Reply:
Fair enough.  I'm just trying to learn here.  Thanks to both of you for the feedback.

I'll commit this change as well.  Let me know if you prefer me to squash the submission into a single commit.